db/migrate/20250414121956_migrate_ovirt_resources.rb

```diff
@@ -1,9 +1,17 @@
 class MigrateOvirtResources < ActiveRecord::Migration[7.0]
   def up
-    ComputeResourceCleaner.new(klass: 'Foreman::Model::Ovirt').run!
+    unless plugin_present?('foreman_ovirt')
+      ComputeResourceCleaner.new(klass: 'Foreman::Model::Ovirt').run!
+    end
   end
 
   def down
     raise ActiveRecord::IrreversibleMigration, "Cannot restore migrated Ovirt data."
   end
+
+  private
+
+  def plugin_present?(gem_name)
+    Gem.loaded_specs.key?(gem_name)
+  end
```

Comment:
Please reuse what we already have for the macro, which avoids using `Gem.loaded_specs`: https://github.com/theforeman/foreman/blob/c28ab6ff1cd0b8fe77b853a5bfbd70ef86508af9/app/services/foreman/renderer/scope/macros/base.rb#L48-L56

Reply:
I probably don't know how to do this, in my [previous attempt](https://github.com/theforeman/foreman/commit/8f45b46304d69f6c84198c93e5cce77b102a2f06) `plugin_present` [caused an error](https://github.com/theforeman/foreman/actions/runs/17097719483/job/48486134618?pr=10657). :confused:

Reply:
You can probably copy the implementation:
```suggestion
  def plugin_present?(name)
    Foreman::Plugin.find(name).present?
  end
```